src/main.cpp

```diff
@@ -2617,7 +2625,9 @@ void txRxProcessing()
 
         int nsam_in_48 = g_soundCard2SampleRate * freedvInterface.getTxNumSpeechSamples()/freedvInterface.getTxSpeechSampleRate();
         assert(nsam_in_48 < 10*N48);
-        while((unsigned)codec2_fifo_free(cbData->outfifo1) >= nsam_one_modem_frame) {
+        
+        int bytesNeededForWrite = nsam_one_modem_frame * (g_soundCard1SampleRate/g_soundCard2SampleRate);
```

Comment:
I'm not sure about this - think it should still be:
```
while((unsigned)codec2_fifo_free(cbData->outfifo1) >= nsam_one_modem_frame) {
```

Outfifo1 is containing frames of modulated modem signals at the g_soundCard2SampleRate sample rate, so I figure we should run around this loop while there is room for one more output modem frame at that rate. 

For example if the modem sample rate is 8000 Hz, the sound card 48000 Hz, and there is 1288 samples/modem frame at 8000 Hz, then `nsam_one_modem_frame` = 48000*1288/8000, the number of samples in one modem frame, at the output sample rate. 

Sorry of I've missed something here ... this code is pretty tricky :thinking:

Reply:
I played with it some more and I think you're right. However, `nsam_one_modem_frame` should use `g_soundCard1SampleRate`--not `g_soundCard2SampleRate`--based on the comments already in the code. In fact, testing it this way seemed to work (16K sample rate for the microphone, 48K for the radio device). I'll update accordingly.

Reply:
Good catch :+1: